instance/ansible.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Ansible - Helper dunctions
+"""
+
+# Imports #####################################################################
+
+import os
+import subprocess
+import yaml
+
+from tempfile import mkdtemp, mkstemp
+
+from django.conf import settings
+
+
+# Logging #####################################################################
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+# Functions ###################################################################
+
+def yaml_merge(yaml_str1, yaml_str2):
+    """
+    Merge the two yaml strings, overriding variables from `yaml_str1` by `yaml_str2`
+    """
+    if not yaml_str2:
+        return yaml_str1
+
+    result_dict = yaml.load(yaml_str1)
+    for key, value in yaml.load(yaml_str2).items():
+        if key in result_dict and isinstance(result_dict[key], dict) and isinstance(value, dict):
```

Comment:
@antoviaque nit: it might backfire when/if there are three levels of nested dicts (root, nested1 and nested2). Not sure if it is ever going to happen, but it might make sense to fix now, as it likely would be quite hard to debug when something actually breaks.